### PR TITLE
[FIX]: Libraries shadowing

### PR DESCRIPTION
**Pull request by imLinguin**
fixes shadowing of some libraries. Thanks @Etaash-mathamsetty for finding this bug

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by Etaash-mathamsetty** [comment]
The issue is not fully fixed, but at least WRC works now :) 
mangohud still doesn't work unfortunately

**Comment by imLinguin** [comment]
This PR should also fix issues with Steam runtime in Heroic flatpak

**Comment by arielj** [comment]
I'm not sure how to review this one cause I don't understand what's the issue it's fixing, can you provide more context in the description @imLinguin ?

cause I also see the branch is named `update-nile` but it's doing other things

**Comment by imLinguin** [comment]
This solves multiple issues with gogdl and nile shadowing the libraries of subprocesses. It also moves all wrappers to be added into `--wrapper` parameter, instead wapping our tools in them (which solves some issues with Steam runtimes in Flatpak too)

Added Nile to gamesdb mapping, fixed how settings detect native game (they checked if native version is available).

**Comment by arielj** [comment]
got it, I understand the update for gogdl and nile are new binaries supporting the `--wrapper` option, right?

changes look good to me, at least what I understand of the code, I've never had the issue so I'm not sure it to actually test a before/after but I trust you and @Etaash-mathamsetty tested this already and works

maybe somebody else with a better understanding about wrappers can also review the code

File: src/backend/config.ts [hunk 1]
```diff
@@ -320,7 +320,9 @@ class GlobalConfigV0 extends GlobalConfig {
       },
       wineCrossoverBottle: 'Heroic',
       winePrefix: isWindows ? '' : defaultWinePrefix,
-      wineVersion: defaultWine
+      wineVersion: defaultWine,
+      enableEsync: true,
+      enableFsync: true
```
Comment (arielj) [inline src/backend/config.ts:325]: this is changing the default to be `true` right? is this intentional?

I'm fine with that, just want to be sure this was the idea

**Comment by imLinguin** [comment]
Yes, there are also improvements for removing part from LD_LIBRARY_PATH which is added by pyinstaller (essentially solution for library shadowing)

File: src/backend/config.ts (hunk 1 above)
Comment (imLinguin) [reply src/backend/config.ts:325]: Yeah, I find it weird that we didn't have esync and fsync as default

**Review by flavioislima — APPROVED** [review]
LGTM

**Comment by flavioislima** [comment]
It is working fine, although now you have conflicts to fix 😢

**Comment by imLinguin** [comment]
will solve them tomorrow. Then I'll also update nile to handle being offline, it bails out currently (story for another PR)

**Comment by flavioislima** [comment]
> will solve them tomorrow. Then I'll also update nile to handle being offline, it bails out currently (story for another PR)

Sure, no problem 👍🏽

**Pull request merged**